AITEWIN ROBOT ESP32-C3 SuperMini

geekgarage opened the pull request:
I added the AITEWIN ROBOT ESP32-C3 SuperMini that's a somewhat clone or derivative of the Adafruit QT Py ESP32-C3

I've tested all IO's and cannot find any faults or improper configuration.

(sorry for redoing PR. The first pin out diagram i found was incorrect so i had to redo all labels in the config, and somehow i managed to delete my existing PR instead of updating it)

Here is the proper Pinout for this seller 
https://www.aliexpress.com/item/1005005757810089.html?spm=a2g0o.order_list.order_list_main.87.37ee18028dkcaE
![Pinout-Back](https://github.com/wokwi/wokwi-boards/assets/12700648/c9d7abc3-d3ce-4385-b3f7-3507b0aa1dc9)

geekgarage committed:
Added AITEWIN ROBOT ESP32-C3 SuperMini

geekgarage committed:
Merge pull request #1 from geekgarage/GeekCustomBoards

Added AITEWIN ROBOT ESP32-C3 SuperMini

geekgarage committed:
Update board.json

Found out dealing with SPI and I2C that the pin out diagram from seller was incorrect. Found another pin out diagram, tested with those description and they both work now

geekgarage committed:
Merge pull request #2 from geekgarage/GeekCustomBoards

Update board.json

geekgarage committed:
Update board.json

Missing extra labels on pin 10, 20 and 21

geekgarage committed:
changed name

geekgarage committed:
Added file in case feature reques gets approved and you need test data (https://github.com/wokwi/wokwi-boards/issues/29)

urish commented:
Thanks for the PR! Sorry for not looking into this earlier, I'm a bit swamped with other stuff. Anyway, I plan to look into it this week and merge if all good

geekgarage commented:
> Thanks for the PR! Sorry for not looking into this earlier, I'm a bit swamped with other stuff. Anyway, I plan to look into it this week and merge if all good

I added some extra files not currently needed in case you think #29 feature request could be something in the long run, then there is some test data to play around with

urish commented:
Thanks! 

Some feedback: 

1. Please replace the "_" with a dash (i.e. `aitewin_robot-esp32-c3-supermini`
2. Please change the pin spacing in board.json so it's an exact multiple of 2.54. It's very already very close to a multiple of 2.54, so it's just a minor adjustment of the coordinates, both in the X and Y axes. 

Note my comments about the CSV / TXT file in #29.

geekgarage committed:
Fix Pin Spacing

geekgarage committed:
Changing Name, adding buy links in JSON, renaming folder, removing CSV and txt

geekgarage commented:
> Thanks!
> 
> Some feedback:
> 
>     1. Please replace the "_" with a dash (i.e. `aitewin_robot-esp32-c3-supermini`
> 
>     2. Please change the pin spacing in board.json so it's an exact multiple of 2.54. It's very already very close to a multiple of 2.54, so it's just a minor adjustment of the coordinates, both in the X and Y axes.
> 
> 
> Note my comments about the CSV / TXT file in #29.

I've done the following:
- Removed the two file (CSV and TXT)
- Renamed the board in the JSON to include manufacturer
- Added a buy section with how i think it could be done. you can just remove it if it won't play nice or you want it done in another way.
- Renamed the folder
- Made sure there is no underscores in files and folders
- Fixed Pin spacing in X and Y so it's dividable with 2,54mm

urish commented:
Thank you! I hope to get to review it by Sunday and merge. Will keep you posted

urish commented on boards/aitewinrobot-esp32c3-supermini/board.json:
```diff
@@ -0,0 +1,67 @@
+    "5V": { "x": 16.702, "y": 3.238, "target": "power(5)" },
+    "GND": { "x": 16.702, "y": 5.778, "target": "GND" },
+    "3V3": { "x": 16.525199, "y": 8.318, "target": "power(3.3)" },
+    "4/GPIO4/A4/SCK": { "x": 16.702, "y": 10.858, "target": "GPIO4" },
+    "3/GPIO3/A3": { "x": 16.702, "y": 13.398, "target": "GPIO3" },
+    "2/GPIO2/A2": { "x": 16.702, "y": 15.938, "target": "GPIO2" },
+    "1/GPIO1/A1": { "x": 16.702, "y": 18.478, "target": "GPIO1" },
+    "0/GPIO0/A0": { "x": 16.702, "y": 21.018, "target": "GPIO0" },
+    "5/GPIO5/A5/MISO": { "x": 1.462, "y": 3.238, "target": "GPIO5" },
+    "6/GPIO6/MOSI": { "x": 1.462, "y": 5.778, "target": "GPIO6" },
+    "7/GPIO7/SS": { "x": 1.462, "y": 8.318, "target": "GPIO7" },
+    "8/GPIO8/SDA": { "x": 1.462, "y": 10.858, "target": "GPIO8" },
+    "9/GPIO9/SCL": { "x": 1.462, "y": 13.398, "target": "GPIO9" },
+    "10/GPIO10": { "x": 1.462, "y": 15.938, "target": "GPIO10" },
+    "20/GPIO20/RX": { "x": 1.462, "y": 18.478, "target": "GPIO20" },
+    "21/GPIO21/TX": { "x": 1.462, "y": 21.018, "target": "GPIO21" }
```
Let's keep the pin names simple. I'm thinking:

1. For "RX/TX" pins, call them as "RX" / "TX"
2. For all the other pins, just the GPIO index (so "4", "3", etc)

This will keep us consistent with the other ESP32 boards (such as the esp32-devkit-c-v4)

geekgarage replied:
While I see your point in relation to other boards, and i will change it, I don't agree at all with simplifying the names.
My reasoning being when you run wires between devices, it can be hard to remember what pin is what and have what functionality, and then you would constantly have to refer back to the diagram. So while i see the aesthetically pleasing by simplifying pin names, you also make it a much worse user experience, by prioritizing form over function.

Let me know if you still think this should be changed and I'll fix it

geekgarage replied:
oh and as a tablet and PC user, i can say that prioritizing the form over function, is just compounded when working on a tablet with a single screen and you have to flip flop between tabs to see the functions of the pins. The user experince just gets even worse, then on top of that, add in if you are a new user to all of this, both arduino and Wokwi, you have enough information to keep track on and start learning.

geekgarage replied:
I've simplyfied the pin name, but i would like you to consider my points about user experience issues

urish commented on boards/aitewinrobot-esp32c3-supermini/board.json:
```diff
@@ -0,0 +1,67 @@
+  "description": "A common ESP32 dev board by AITEWIN ROBOT (Derived from Adafruit QT Py ESP32-C3)",
+  "author": "GeekGarage",
+  "mcu": "esp32-c3",
+  "fqbn": "esp32:esp32:esp32c3",
```
Looks like this one is a better fit: https://github.com/espressif/arduino-esp32/commit/50f436c6fc3d525d08a81f6e0f222b88a7e6cc7d

It wasn't released yet, so let's keep the current value for now

geekgarage replied:
I'm guessing we are talking about the MCU and FQBN ?

geekgarage replied:
since we are keeping current values I'm guessing it is fixed for now. Let me know when the board is released and I will update

urish commented on boards/aitewinrobot-esp32c3-supermini/board.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "Aitewin Robot ESP32-C3 SuperMini",
+  "version": 1.8,
```
version should be an integer - set it to 1, since this is the first version that will be released.

urish commented on boards/aitewinrobot-esp32c3-supermini/board.json:
```diff
@@ -0,0 +1,67 @@
+        "C": "GND"
+      }
+    },
+    {
+      "id": "io8led",
+      "x": 13.599,
+      "y": 12.885,
+      "type": "0603",
+      "color": "blue",
+      "radius": 3.0,
+      "pins": {
+        "A": "8",
+        "C": "GND"      
+      }
+    }
```
The LED doesn't work - but it might work after changing the pin names, as suggested above

geekgarage replied:
I'll fix it when I get a response on the first one as they are depended on each other :)

geekgarage replied:
it's fixed

urish reviewed (changes requested):
Thanks again!

geekgarage committed:
Update board.json

change version

geekgarage committed:
Update board.json

Temp update led pin name for the onboard GPIO8 LED

geekgarage committed:
Update board.json

Updated the incorrect field regarding onboard GPIO8 LED

geekgarage committed:
Merge branch 'wokwi:main' into main

geekgarage committed:
Update board.json

SImplifying the pin names as requested for merge into Wokwi.
NOTE: This will make for a worse user experience IMO

urish commented:
Thanks again for contributing this board! It's now available in Wokwi :)

geekgarage commented:
> Thanks again for contributing this board! It's now available in Wokwi :)

Sweet, how would i go about using it then. i tried changing the board to "aitewinrobot-esp32c3-supermini" as that's what it is named but nothing happens. So I'm a bit confused as to how I would go about using it in my projects on Wokwi :P

urish commented:
Yeah, it should be `board-aitewinrobot-esp32c3-supermini`. Here's the obligatory blink: https://wokwi.com/projects/389262377969018881

geekgarage commented:
aaah i forgot the "board-" part
